src/Entities/Entity.cpp

```diff
@@ -1835,6 +1702,131 @@ void cEntity::TeleportToCoords(double a_PosX, double a_PosY, double a_PosZ)
 
 
 
+bool cEntity::MoveToWorld(cWorld & a_NewWorld, const Vector3d & a_NewPosition)
+{
+	auto PreviousDimension = GetWorld()->GetDimension();
+	if (OnPreWorldTravel(a_NewWorld))
+	{
+		OnPostWorldTravel(PreviousDimension, a_NewPosition);
+		SetPosition(a_NewPosition);  // Just in case :)
+		return true;
+	}
+	return false;
+}
+
+
+
+
+
+bool cEntity::MoveToWorld(const AString & a_WorldName, const Vector3d & a_NewPosition)
+{
+	auto World = cRoot::Get()->GetWorld(a_WorldName);
+	if (World == nullptr)
+	{
+		LOG("%s: Couldn't find world \"%s\".", __FUNCTION__, a_WorldName.c_str());
+		return false;
+	}
+
+	return MoveToWorld(*World, a_NewPosition);
+}
+
+
+
+
+
+bool cEntity::OnPreWorldTravel(cWorld & a_NewWorld)
+{
+	ASSERT(IsTicking());
+
+	if (GetWorld() == &a_NewWorld)
+	{
+		// Don't move to same world
+		return false;
+	}
+
+	//  Ask the plugins if the player is allowed to changing the world
+	if (cRoot::Get()->GetPluginManager().CallHookEntityChangingWorld(*this, a_NewWorld))
+	{
+		// A Plugin doesn't allow the player to changing the world
+		return false;
+	}
+
+	// Prevent further ticking in this world
+	SetIsTicking(false);
+
+	GetWorld()->QueueTask(
+		[this, &a_NewWorld](cWorld & a_CurrentWorld)
+		{
+			auto Entity = GetParentChunk()->GetAssociatedEntityPtr(*this);
+
+			class Callback : public cEntityCallback
+			{
+			public:
+				Callback(const std::weak_ptr<cPawn> & a_Pawn) :
+					m_Pawn(a_Pawn)
+				{
+				}
+
+				virtual bool Item(cEntity * a_Entity) override
+				{
+					if (!a_Entity->IsMob())
+					{
+						return false;
+					}
+
+					auto Monster = static_cast<cMonster *>(a_Entity);
+					if (std::equal_to<std::weak_ptr<cPawn>>()(Monster->GetWeakTargetPtr(), m_Pawn))
+					{
+						Monster->SetTarget(nullptr);
```

Comment:
>I was thinking that since entity world moving is typically rare, more benefits would be seen by not needing to maintain a list of targeting entities, and moving that processing to a one-time iteration when moving worlds.

You are correct. When I first made my comment, I was still fixated on the old code, in which destruction also required iterating pointers. World travel is rare, destruction is not.

> Dunno about ForEachEntity - if it doesn't catch all then I have a bug :)
> Where should I perform this copy, and to what object?

Entities in invalid chunks are missed. See `cChunkMap::cChunkLayer::ForEachEntity`, and note the `&& m_Chunks[i]->IsValid();`. There are two possible resolutions:

1. Establishing a rigid policy on what we expect of invalid chunks to do and not to do, and consistently change all relevant functions such as `cChunkMap::cChunkLayer::ForEachEntity`, and make `ForEachEntity` iterate everything.

2. I was thinking that perhaps copying the entity to the new world and then destroying the entity in the old world rids you of this whole iteration.